tests/lib/rules/no-new-func.js

```diff
@@ -55,6 +55,34 @@ ruleTester.run("no-new-func", rule, {
                 type: "CallExpression"
             }]
         },
+        {
```

Comment:
Can you also add these `valid` test cases to confirm that the two false positives from https://github.com/eslint/eslint/pull/14860#discussion_r696758147 have been fixed:

```js
"foo[Function]()",
"foo(Function.bind)"
```

Reply:
@mdjermanovic Could you please provide a suggestion or code example?

Reply:
@mdjermanovic to me it is clear how these tests should look. Would you mind pushing a commit to add those, if needed?

Reply:
Done, I also added a test from https://github.com/eslint/eslint/pull/14860#discussion_r702853939

Reply:
Thanks very much. 
Are you planning to release a patch including this PR in near future?

Reply:
We will certainly include this change in the next prerelease (https://github.com/eslint/eslint/issues/15059) if it gets merged before we start the release.

I think https://github.com/eslint/eslint/pull/14860#discussion_r702834472 is the only thing left to address before merging this PR.